### PR TITLE
[21.01] Fix bug with Pulsar and $__tool_directory__

### PR DESCRIPTION
**Pull request by abretaud**
## What did you do? 
- Added a test case for Pulsar issue https://github.com/galaxyproject/pulsar/issues/249

## Why did you make this change?
(Cite Issue number OR provide rationalization of changes if no issue exists)
(If fixing a bug, please add any relevant error or traceback)
The bug is not yet fixed, but at least it should be detected by this test now

## How to test the changes? 
(select the most appropriate option; if the latter, provide steps for testing below)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).

### CODE REVIEW

**Comment by abretaud** [comment]
Ok, I've just pushed a patch that fixes the problem on my test instance and makes the test pass locally.
Maybe not super elegant, but at least I understand what's going on now :) (=pulsar needs to know when the command factory decides to externalize the command line in a tool_script.sh file)

File: lib/galaxy/jobs/runners/pulsar.py [hunk 1]
```diff
@@ -378,6 +378,9 @@ def queue_job(self, job_wrapper):
                     remote_pulsar_app_config.update(yaml.safe_load(fh))
             job_directory_files = []
             config_files = job_wrapper.extra_filenames
+            if externalized_command:
+                config_files.append(externalized_command)
```
Comment (mvdbeek) [inline lib/galaxy/jobs/runners/pulsar.py]: I think you should have the same effect if you add `config_files.append(tool_script)` after line 387

**Review by mvdbeek — COMMENTED** [review]
Nice find!

File: lib/galaxy/jobs/runners/pulsar.py (hunk 1 above)
Comment (abretaud) [reply lib/galaxy/jobs/runners/pulsar.py]: Ah yes, much simpler and it works :)

**Review by mvdbeek — APPROVED** [review]

**Comment by abretaud** [comment]
Is it expected that integration tests take so long to run on GA?

**Comment by abretaud** [comment]
Ok, green now \o/

**Comment by mvdbeek** [comment]
Thanks so much @abretaud!

**Pull request merged**